src/kinematics/DiffDriveKinematics.cpp

```diff
@@ -49,4 +50,55 @@ pose_t DiffDriveKinematics::getNextPose(const wheelvel_t& wheelVel, const pose_t
 										double dt) const {
 	return pose + getPoseUpdate(wheelVel, pose(2), dt);
 }
+
+pose_t DiffDriveKinematics::ensureWithinWheelSpeedLimit(
+	PreferredVelPreservation preferredVelPreservation, double xVel, double thetaVel,
+	double maxWheelSpeed) const {
+	auto [lVel, rVel] = robotVelToWheelVel(xVel, thetaVel);
+	double calculatedMaxWheelVel = std::max(std::abs(lVel), std::abs(rVel));
+	if (calculatedMaxWheelVel > maxWheelSpeed) {
+		switch (preferredVelPreservation) {
+			case PreferredVelPreservation::Proportional: {
+				xVel *= maxWheelSpeed / calculatedMaxWheelVel;
+				thetaVel *= maxWheelSpeed / calculatedMaxWheelVel;
+			}
```

Comment:
Missing a break?

Reply:
fixed. also added const qualifiers to the variables declared at the top of the function that should not change